utils/errors.go

```diff
@@ -59,3 +59,8 @@ func typeStr(of interface{}) string {
 func NewUnimplementedInterfaceError(expected, actual interface{}) error {
 	return errors.Errorf("expected implementation of %s but got %T", typeStr(expected), actual)
 }
+
+// InvalidIntegerValue is used when an integer value is out of bound or not valid.
+func InvalidIntegerValue(val int) error {
+	return errors.Errorf("can not validate integer value %d in config", val)
```

Comment:
a more specific error message is needed. This doesn't tell me what wrong in terms of baud rate.